components/brave_today/browser/brave_news_controller.cc

```diff
@@ -343,16 +343,16 @@ void BraveNewsController::GetDisplayAd(GetDisplayAdCallback callback) {
         // TODO(petemill): brave_ads seems to use mojom, perhaps we can receive
         // and send to callback the actual typed mojom struct from brave_ads?
         auto ad = mojom::DisplayAd::New();
-        ad->uuid = *ad_data.FindString("uuid");
-        ad->creative_instance_id = *ad_data.FindString("creativeInstanceId");
-        if (const auto* value = ad_data.FindString("ctaText"))
+        ad->uuid = *ad_data->FindString("uuid");
```

Comment:
Not connected to the current PR:
Seems like DCHECK was missing for `FindString()` call in previous code here and below. 
Another variant is to convert `ad_data` to mojom entity as per comment above.

Reply:
ad_data will be refactored in the future to a mojom struct; however, we first need to refactor AdInfo.